task/monorail-server/src/bsp/sidecar_bcd.rs

```diff
@@ -472,6 +500,15 @@ impl<'a, R: Vsc7448Rw> Bsp<'a, R> {
             }
         }
 
+        if let VLanMode::UnlockedUntil(t) = self.vlan_mode {
+            let now = userlib::sys_get_timer().now;
+            if now >= t {
+                ringbuf_entry!(Trace::AutomaticLock);
+                // Discard the error, because it's already logged in lock_vlans
+                let _ = self.lock_vlans();
+            }
+        }
```

Comment:
I'm not familiar with the full monorail setup, when is `wake` getting called?

Reply:
Every 500 ms, per [`WAKE_INTERVAL`](https://github.com/oxidecomputer/hubris/blob/24abf7292491069dc87e1e5a2930ab5822d26abb/task/monorail-server/src/bsp/sidecar_bcd.rs#L19-L20)